src/storage/watermark.py

```diff
@@ -16,71 +23,71 @@ def draw_text_with_outline(draw, position, text, font, text_colour, outline_colo
-            img_w - margin,
-            margin + text_height,
-        ),  # Top-right
-        (
-            margin,
-            img_h - text_height - margin,
-            margin + text_width,
-            img_h - margin,
-        ),  # Bottom-left
-        (
-            img_w - text_width - margin,
-            img_h - text_height - margin,
-            img_w - margin,
-            img_h - margin,
-        ),  # Bottom-right
-    ]
-
-    min_detail = float("inf")
+    # Define dictionary with the coordinates of each corner with some margin applied
+    corners = {
+        "top_left": (margin, margin),
+        "top_right": (img_w - text_width - margin, margin),
+        "bottom_left": (margin, img_h - text_height - margin),
+        "bottom_right": (img_w - text_width - margin, img_h - text_height - margin),
+    }
+    # Initialize with infinity for comparison
+    least_detailed_score = float("inf")
+    # To store the x, y coordinates for watermark
     selected_corner = None
-
-    for corner in corners:
-        sector = gray_img.crop(corner)
-        stat = ImageStat.Stat(sector)
-        mean = stat.mean[0]  # Mean pixel value in the sector
-        w_border, b_border = 255, 0
-        min_diff_for_corner = min(abs(mean - w_border), abs(mean - b_border))
-
-        if min_diff_for_corner < min_detail:
-            min_detail = min_diff_for_corner
-            selected_corner = corner[:2]  # Keep only the starting coordinates
-    return selected_corner
+    selected_wm_colour = None
+    # Iterate through each corner to find the one with the least detail background
+    for _, (x, y) in corners.items():
+        # Crop the image around the current corner
+        # according to the text bounding box size
+        sector = gray_img.crop((x, y, x + text_width, y + text_height))
+        # Calculate mean intensity of the cropped area
+        mean_intensity = ImageStat.Stat(sector).mean[0]
+        # Determine how close the mean intensity is to either black (0) or white (255)
+        # indicates how uniform the background of the selected sector is
+        distance_to_extreme = min(abs(0 - mean_intensity), abs(255 - mean_intensity))
+        # Update if this corner has a lower score (closer to black or white)
+        if distance_to_extreme < least_detailed_score:
+            least_detailed_score = distance_to_extreme
+            selected_corner = (x, y)
+            # Selecting watermark colour based on mean intensity
+            selected_wm_colour = (0, 0, 0) if mean_intensity > 178 else (255, 255, 255)
+
+    return selected_corner, selected_wm_colour
 
 
 def check_font(font_path, font_family, image, width_ratio):
     fontsize = image.size[0] * width_ratio // 2
     font_file = Path(font_path) / font_family
-    font = ImageFont.truetype(str(font_file), fontsize)
+    try:
+        font = ImageFont.truetype(str(font_file), fontsize)
+        logging.info(f"Loaded font from {font_file}.")
+    except IOError:
```

Comment:
We need to work with the font provided in the repo. It will be not cool if at some point we'd start to set watermarks with default font with uncontrolled font size and other params. We need to raise an error if a font fails to load.